features/environment.py

```diff
@@ -27,19 +27,10 @@
 
 USERDATA_BLOCK_AUTO_ATTACH = """\
 #cloud-config
-write_files:
-  # TODO(drop path: /usr/bin/ua when 25.0 is in Ubuntu PRO images)
-  - path: /usr/bin/ua
-    content: |
-        #!/bin/bash
-        DATE=`date -u`
-        echo "$DATE: exec ua $@" >> /root/ua-runs
-    permissions: '0755'
-  - path: /etc/ubuntu-advantage/uaclient.conf
-    content: |
-      features:
-         disable_auto_attach: true
```

Comment:
I switched to bootcmd from write_files for these reasons:
1. bootcmd is earlier than write_files processing
2. the bootcmd still disables /usr/bin/ua without also dumping changes into uaclient.conf
3. in write_files we are modifying uaclient.conf and when you install ubuntu-advantage-tools deb on that modified instance you get a prompt saying that there are local config changes present in uaclient.conf, do you want to keep maintainer's version or your own. This breaks PRO tests because we don't provide DEBIAN_FRONTEND=noninteractive which would have defaulted the answer.

Reply:
But my concern is about what happens after we install the uaclient deb that contains the code we are testing, either from daily or from deb built by the PR. When we install that deb on the machine, the `ua` override will be gone.

Since we will no longer have the `disable_auto_attach` config set up, we will end creating a pro image that will already be attached before we run our tests, right ?

Also, regarding the missing DEBIAN_FRONTEND. I don't know if I am missing something, but if that was the case, shouldn't the pro fail right now ?

Reply:
@lucasmoura I suppose it should have failed, I think I may have misinterpreted the error logs due to other build failures. Per our conversation today I did separate a USERDATA_NO_AUTO_ATTACH_TEST config which we now provide to pro test  instances at launch time which sets the appropriate uaclient.conf disable_auto_attach config override. I also separate the build vms USERDATA_NO_AUTO_ATTACH_IMG which is the simple and destructive bootcmd which kills /usr/bin/ua. 

This way we get no errors or warning logs on build images when installing ubuntu-*debs and seeing local uaclient.conf changes, and our USERDATA_NO_AUTO_ATTACH_TEST ensures that integration test launches on pro machines still avoid auto-attach on first boot.